src/System.IdentityModel.Tokens/Utility.cs

```diff
@@ -65,5 +65,37 @@ internal static string SerializeAsSingleCommaDelimitedString(IEnumerable<string>
 
             return sb.ToString();
         }
+
+        public static bool IsHttps(string address)
+        {
+            if (string.IsNullOrEmpty(address))
+            {
+                return false;
+            }
+
+            try
+            {
+                Uri uri = new Uri(address);
+                return IsHttps(new Uri(address));
+            }
+            catch (UriFormatException)
+            {
+                return false;
+            }
+
+        }
+
+        public static bool IsHttps(Uri uri)
+        {
+            if (uri == null)
+            {
+                return false;
+            }
+#if DNXCORE50
+            return uri.Scheme == "https"; //Uri.UriSchemeHttps is internal in dnxcore
```

Comment:
It is hardcoded to be lower case: https://github.com/dotnet/corefx/blob/41e203011152581a6c65bb81ac44ec037140c1bb/src/System.Private.Uri/src/System/UriSyntax.cs#L121. But adding .ToLower() seems a good thing to do.